src/spikeinterface/extractors/nwbextractors.py

```diff
@@ -590,173 +874,6 @@ def __init__(
-        The number of timestamp samples used for estimating the sampling rate. This is relevant
-        when the 'rate' attribute is not available in the ElectricalSeries.
-    stream_mode : "fsspec", "ros3", "remfile", or None, default: None
-        Determines the streaming mode for reading the file. Use this for optimized reading from
-        different sources, such as local disk or remote servers.
-    cache: bool, default: False
-        Indicates whether to cache the file locally when using streaming. Caching can improve performance for
-        remote files.
-    stream_cache_path: str, Path, or None, default: None
-        Specifies the local path for caching the file. Relevant only if `cache` is True.
-    use_pynwb: bool, default: False
-        Uses the pynwb library to read the NWB file. Setting this to False, the default, uses h5py
-        to read the file. Using h5py can improve performance by bypassing some of the PyNWB validations.
-    electrical_series_location: str or None, default: None
-        This parameter is only used when `use_pynwb=False`.
-        Specifies the direct path to the ElectricalSeries object within the NWB file,
-        e.g., 'acquisition/ElectricalSeries/my_electrical_series'.
-
-        Precedence:
-        - If provided, this parameter directly locates the ElectricalSeries, enhancing
-        data loading speed.
-        - If absent but `electrical_series_name` is specified, the extractor searches
-        for an ElectricalSeries with the given name across the NWB file.
-        - If both this parameter and `electrical_series_name` are unspecified, and there
-        is only one ElectricalSeries in the file, that series is automatically selected
-        otherwise an error is raised.
-
-    Returns
-    -------
-    recording : NwbRecordingExtractor
-        The recording extractor for the NWB file.
-
-    Examples
-    --------
-    Run on local file:
-
-    >>> from spikeinterface.extractors.nwbextractors import NwbRecordingExtractor
-    >>> rec = NwbRecordingExtractor(filepath)
-
-    Run on s3 URL from the DANDI Archive:
-
-    >>> from spikeinterface.extractors.nwbextractors import NwbRecordingExtractor
-    >>> from dandi.dandiapi import DandiAPIClient
-    >>>
-    >>> # get s3 path
-    >>> dandiset_id, filepath = "101116", "sub-001/sub-001_ecephys.nwb"
-    >>> with DandiAPIClient("https://api-staging.dandiarchive.org/api") as client:
-    >>>     asset = client.get_dandiset(dandiset_id, "draft").get_asset_by_path(filepath)
-    >>>     s3_url = asset.get_content_url(follow_redirects=1, strip_query=True)
-    >>>
-    >>> rec = NwbRecordingExtractor(s3_url, stream_mode="fsspec", stream_cache_path="cache")
-    """
-
-    extractor_name = "NwbRecording"
-    mode = "file"
-    name = "nwb"
-
-    def __new__(
-        cls,
-        file_path: str | Path | None = None,  # provide either this or file
-        electrical_series_name: str | None = None,
```

Comment:
We probably should  deprecate  ` electrical_series_name`  instead of removing it outright.

Reply:
agreed

Reply:
and in case it's used we can automatically set `electrical_series_path=f"acquisition/{electrical_series_name}"`

Reply:
is that consistent with the old behavior? The old behavior looks for the name everywhere not just in acquisition I think.

Reply:
probably not, but it was ambiguous. I think this is a better solution. If it fails, it will print the available paths and users can use the paths instead

Reply:
The downside of this is that you might break working code. If you are willing to do that I would rather just thrown an error when someone passess the argument, do a serach and give them the possible paths they can enter.

Reply:
Mmm now that I think about it your proposed solution does avoids breaking some working code (not all of it) which seems better.